force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,60 @@
+trigger AccountTrigger on Account (before insert, after insert) {
+
+  if (Trigger.isBefore) {
+
+    for (Account a : Trigger.new) {
+
+      if (Trigger.isInsert) {
+        // Question 1
+        if (a.Type == Null) {
+          a.Type = 'Prospect';
+        }
+        // Question 2
+        if (a.ShippingStreet != Null) {
+          a.BillingStreet = a.ShippingStreet;
+        }
+
+        if (a.ShippingCity != Null) {
+          a.BillingCity = a.ShippingCity;
+        }
+
+        if (a.ShippingState != Null) {
+          a.BillingState = a.ShippingState;
+        }
+
+        if (a.ShippingPostalCode != Null) {
+          a.BillingPostalCode = a.ShippingPostalCode;
+        }
+
+        if (a.ShippingCountry != Null) {
+          a.BillingCountry = a.ShippingCountry;
+        }
+        //Question 3
+        if (a.Phone != Null && a.Website != Null && a.Fax != Null) {
+          a.Rating = 'Hot';
+        }
+
+      } 
+
+    }
+
+  }
+
+  if (Trigger.isAfter) {
+    if (Trigger.isInsert) {
+      //Question 4
+      List<Contact> newContacts = new List<Contact>();
+
+      for (Account a : Trigger.new) {
+
+        newContacts.add(new Contact(AccountId = a.Id, LastName = 'DefaultContact', Email = 'default@email.com'));
```

Comment:
This will create a new contact on every update of the account. You probably want to check if the contact is not there already. The directions in the homework are unclear, but in a realistic scenario, you probably would want to get more clarification.